device_calendar/example/lib/presentation/pages/calendar_event.dart

```diff
@@ -100,23 +99,30 @@ class _CalendarEventPageState extends State<CalendarEventPage> {
         if (_event.recurrenceRule.endDate != null) {
           _recurrenceRuleEndType = RecurrenceRuleEndType.SpecifiedEndDate;
           _recurrenceEndDate = _event.recurrenceRule.endDate;
-          _recurrenceEndTime = TimeOfDay.fromDateTime(_recurrenceEndDate);
         }
-        _daysOfTheWeek =
-            _event.recurrenceRule.daysOfTheWeek ?? List<DayOfTheWeek>();
-        _daysOfTheMonth = _event.recurrenceRule.daysOfTheMonth ?? List<int>();
-        _monthsOfTheYear = _event.recurrenceRule.monthsOfTheYear ?? List<int>();
-        _weeksOfTheYear = _event.recurrenceRule.weeksOfTheYear ?? List<int>();
-        _setPositions = _event.recurrenceRule.setPositions ?? List<int>();
+
+        _isByDayOfMonth = _event.recurrenceRule.setPositions == null ? true : false;
+        _daysOfTheWeek = _event.recurrenceRule.daysOfTheWeek ?? List<DayOfTheWeek>();
```

Comment:
I think it's a good practive to avoid using articles in the names of variables and classes (i.e. lose `the` in the `_daysOfTheWek`, `DayOfTheWeek`). See a SO thread about this https://stackoverflow.com/a/733996/510627

Reply:
All those were already defined like it when I started so I didn't touch them. Do you want me to change all of them?

Reply:
I think that it would be nice, if that's not too much problems. 

I would suggest though to do that in a completely separate PR to avoid clutter with renaming refactoring changes.